frontend/src/lib/components/Popup/Popup.tsx

```diff
@@ -80,6 +83,17 @@ export function Popup({
                       requires: ['computeStyles'],
                   }
                 : {},
+            !!maxWidth
+                ? {
+                      name: 'maxWidth',
+                      enabled: true,
+                      fn: ({ state }) => {
+                          state.styles.popper.width = maxWidth
+                      },
+                      phase: 'beforeWrite',
+                      requires: ['computeStyles'],
+                  }
+                : {},
```

Comment:
This shouldn't really be a part of the JS – let's keep the styling in CSS. This is also how `max-width` of `Popup`s is controlled elsewhere (see `SitePopover`).